core/modules/memman/MemFile.cc

```diff
@@ -107,6 +107,21 @@ MemFile::MLResult MemFile::memLock() {
         return nilResult;
     }
 
+    // TODO: Find a better solution for systems where mmap is not viable, which
+    // manifests as no bytes being locked but bytes are reserved.
+    // Fake mmap as the configuration isn't working properly.
+    // At this point, there is enough free space, it was not a flex table,
+    // but mmap failed. So,this will fake it being a flexilock file and
+    // reserve space for it.
+    if (_memInfo.size() < freeBytes) {
```

Comment:
I'm not happy with it either. That's why it starts out with todo. However, without this change, memman doesn't work well on the cluster at all (queries on large tables go from 6 hours to 8 hours). I think we will need something like this for use on systems where the workers don't have enough memory and/or can't be easily configured so that mmap works.